packages/react/src/components/Combobox/ComboboxOption.tsx

```diff
@@ -113,10 +123,23 @@ const ComboboxOption = forwardRef<HTMLLIElement, ComboboxOptionProps>(
           ? propValue
           : comboboxOptionRef.current?.innerText;
 
-      if (selectedValue === comboboxValue) {
-        setFormValue(
-          typeof formValue === 'undefined' ? comboboxValue : formValue
-        );
+      if (!multiselect) {
+        if (selectedValue === comboboxValue) {
+          setFormValue(
+            typeof formValue === 'undefined' ? comboboxValue : formValue
+          );
+        }
+      } else {
+        if ((selectedValue as ComboboxValue[]).includes(comboboxValue)) {
+          setFormValue((prev) => {
+            const formValueToSet =
+              typeof formValue === 'undefined' ? comboboxValue : formValue;
+
+            if (!prev?.includes(formValueToSet)) {
+              return prev ? prev + ',' + formValueToSet : formValueToSet;
+            }
+          });
+        }
```

Comment:
This is worth discussing, I'm not sure what the best approach is (and it needs to be addressed as part of the RFC). There's 2 distinct approaches we could take:

- Comma delimited form values
- Individual form values

It appears you've taken the first approach, but what if the value itself includes a comma? It can introduce challenges since the server will need to be aware of the specific format to parse by and be able to translate comma delimited values correctly.

Alternatively, each value itself could be a set of hidden fields that map to the same name:

```html
<input type="hidden" name={name} value={value1} />
<input type="hidden" name={name} value={value2} />
<input type="hidden" name={name} value={value3} />
```

I feel like this approach is more scalable and avoids having to directly parse the submitted values. Thoughts?

Reply:
I really like this approach, thanks for suggestion